src/OpenApi/src/OpenApiGenerator.cs

```diff
@@ -328,7 +328,23 @@ private static void GenerateDefaultResponses(Dictionary<int, (Type?, MediaTypeCo
 
     private List<OpenApiTag> GetOperationTags(MethodInfo methodInfo, EndpointMetadataCollection metadata)
     {
-        var tags = metadata.GetMetadata<ITagsMetadata>();
+        var metadataList = metadata.GetOrderedMetadata<ITagsMetadata>();
+
+        if (metadataList.Count > 0)
+        {
+            var tags = new List<OpenApiTag>();
+
+            foreach (var metadataItem in metadataList)
+            {
+                foreach (var tag in metadataItem.Tags)
+                {
+                    tags.Add(new OpenApiTag() { Name = tag });
```

Comment:
I had a lot more fun writing the `.Aggregate` version of this, but sanity prevailed 😆